### DISCUSSION TITLE
How do I globally turn on echo=true when generating jupyter output?

### DISCUSSION THREAD

**Discussion by jlperla** [quarto-dev/quarto-cli#7090] [Q&A] [answered]
### Description

I have a bunch of revealjs slides. Occasionally I have blocks like
    
    ```{python}
    #| echo: false
    a = 2
    ```

in cases where I don't want the output to show in the slides.  But when I do `quarto render myfile.qmd --to ipynb` I want all of the code blocks to display, or else the jupyter notebook wouldn't be functional.  How do I force it to echo all code in that case without manually search/replace all of my files?

**Comment by cscheid** [comment, marked as answer]
I think the easiest way to do it is the other way around. I believe we have `echo: false` by default in presentation formats, so this should do what you need to.

In case the defaults don't work for you, then I think project profiles are what you're looking for: https://quarto.org/docs/projects/profiles.html

**Reply by mcanouil** [reply]
Indeed, `echo: false` is the default for `revealjs` format.
<img width="762" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/0a5696cd-bd7e-4f4f-bea0-9e7f5e3c1763">

Note that you can enable `echo: true` from the command line.

```bash
quarto render index.qmd --metadata echo:true
```

**Comment by jlperla** [comment]
That did it.  Left off the `echo: false` for anything I don't want echo'd in revealjs but want echoed in ipynb, and it had the intended behavior.
